LambdaEngine/Include/ECS/ComponentManager.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "ComponentArray.h"
+#include "Component.h"
+#include "Defines.h"
+
+namespace LambdaEngine
+{
+	class ComponentManager
+	{
+	public:
+		DECL_UNIQUE_CLASS(ComponentManager);
+		ComponentManager();
+		~ComponentManager();
+
+		template<typename Comp>
+		void RegisterComponentType();
+
+		template<typename Comp>
+		Comp& AddComponent(Entity entity, Comp component);
+
+		template<typename Comp>
+		void RemoveComponent(Entity entity);
+
+		template<typename Comp>
+		Comp& GetComponent(Entity entity);
```

Comment:
As mentioned verbally, I think we should have an `GetArray` function. This would allow systems to cache the lookup of the component array.